### ISSUE TITLE
Question: how to pass namespace defintion to xpath filter?

### ISSUE THREAD

**Issue by rastislavszabo** [ncclient/ncclient#166]
Hi,

I would like to pass an xpath filter to the `get` method. It should be the equivalent of this XML request:

```xml
<rpc xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="28">
  <get>
    <filter type="subtree">
      <turing-machine xmlns="http://example.net/turing-machine">
        <transition-function />
      </turing-machine>
    </filter>
  </get>
</rpc>
```

I wanted to use something like:
```python
reply = connection.get(filter=("xpath", "/tm:turing-machine/tm:transition-function"))
```

This produces an invalid request, since `tm` namespace is not defined anywhere:
```xml
<nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:6f210162-f037-47ee-93cd-b3034b0100ac">
  <nc:get>
    <nc:filter type="xpath" select="/tm:turing-machine/tm:transition-function"/>
  </nc:get>
</nc:rpc>
```

(Netopeer2 server returns an error: `XML namespace with prefix "tm" not defined.`)

Correct request would look like (this is how `netopeer2-cli` produces it):
```xml
<rpc xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="30">
  <get>
    <filter type="xpath" xmlns:tm="http://example.net/turing-machine" select="/tm:turing-machine/transition-function"/>
  </get>
</rpc>
```

How this could be achieved with `ncclient`?

**Comment by Nitori-** [comment]
Would this work (I don't have a device to test with right now)?

    reply = connection.get(filter='<filter type="xpath" xmlns:tm="http://example.net/turing-machine" select="/tm:turing-machine/transition-function"/>')

Seems like it should work based on what I see here: https://github.com/ncclient/ncclient/blob/master/ncclient/operations/util.py#L61

**Comment by rastislavszabo** [comment]
It actually needs to be:
```
reply = connection.get(filter='<nc:filter type="xpath" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:tm="http://example.net/turing-machine" select="/tm:turing-machine/transition-function/delta" />')
```
works, but isn't very convenient. In other words, `xpath` filter type (http://ncclient.readthedocs.io/en/latest/manager.html#filter-params) is not usable, one needs to use `subtree` filter to pass a xpath filter :-(

**Comment by fredgan** [comment]
Hi @einarnn ,
This issue is resolved by PR #405, so it can be closed :)

**Comment by einarnn** [comment]
Resolved by #405
